Improve command line flags and usage message

knrafto opened the pull request:
Resolves #195.

Here's the new help message:
```
usage: python3 ok [--help] [options]

ok is an autograder that you can use to run tests, back up your work, and
submit assignments.

You can run all tests with

    python3 ok

There are several "options" you can give ok to modify its behavior. These
options generally have both a short form (preceded by a single dash, like -q)
or a long form (preceded by two dashes, like --question). This is similar to how
many other command line applications accept options. These options can be mixed
and matched in any order. The options are listed in full below, but we'll
describe some of the more common ones here.

To test a specific question, use the -q (or --question) option with the name of
the question:

    python3 ok -q foo
    python3 ok -q 12

By default, only tests that fail will appear. If you want to see the results
from all tests, you can use the -v (or --verbose) option:

    python3 ok -q foo -v

To start an interactive interpreter after a failed test for debugging, use the
-i (or --interactive) option:

    python3 ok -q foo -i

By default, after each test run ok will attempt to back up your work to the
server. To run the tests without any network access, use the --local option:

    python3 ok -q foo --local

To submit the assignment after you're done, use the --submit option:

    python3 ok --submit

Finally, to log out and log in under a different email, use --authenticate:

    python3 ok --authenticate

Visit https://okpy.org to view your backups and submissions.

optional arguments:
  -h, --help            show this help message and exit

running tests:
  -q QUESTION, --question QUESTION
                        run tests for a specific question
  -u, --unlock          unlock tests interactively
  -i, --interactive     start the Python interpreter after a failed test
  -v, --verbose         show all tests, not just passing tests
  --all                 run tests for all questions in config file
  --submit              submit the assignment
  --backup              attempt to reliably backup your work
  --revise              submit composition revision
  --restore             restore assignment from an earlier backup
  --timeout TIMEOUT     set the timeout duration (in seconds) for running
                        tests

experiment options:
  --no-hints            do not give hints
  --hint                give a hint (if available)
  --style               run AutoStyle feedback system

debugging options:
  --version             print the version number and exit
  --tests               display a list of all available tests
  --debug               show debugging output

grading options:
  --lock                lock the tests in a directory
  --score               score the assignment
  --score-out SCORE_OUT
                        write scores to a file
  --config CONFIG       use a specific configuration file
  --export              download all submissions for the current assignment
  --latest              when used with --export, download latest submissions
                        instead of final submissions

server options:
  --local               disable any network activity
  --server SERVER       set the server address
  --authenticate        authenticate, ignoring previous authentication
  --get-token           get ok access token
  --insecure            use http instead of https
  --no-update           do not check for ok updates
  --update              update ok and exit
```

I'm worried the initial message might be too long, but I tried to be friendly to those who have never used a command line before.

Nitpicks welcome!

Sumukh commented on client/cli/ok.py:
```diff
@@ -27,82 +67,85 @@
-    parser.add_argument('-u', '--unlock', action='store_true',
+
+    testing = parser.add_argument_group('running tests')
+    testing.add_argument('-q', '--question', type=str, action='append',
```
typo: specifi

Sumukh commented:
A common request is how to login under a different email. We might want add that to the blurb (it's currently at the bottom as one of many flags) 

I think most of the info from http://cs61a.org/articles/using-ok.html is here. 

Another related idea: 
Have a sample assignment with an unlocking test suite that serves as an ok walkthrough. Instructors could use that as their intro to using Ok assignment (for 61A: that would be in Lab 0)

knrafto commented:
I've fixed a couple typos and updated the PR description. I did take inspiration from the "Using OK" article, so hopefully we can direct students to this help message instead.

Sumukh commented on client/cli/ok.py:
```diff
@@ -27,82 +71,85 @@
+                        default=sys.stdout, help="write scores to a file")
+    grading.add_argument('--config', type=str,
+                        help="use a specific configuration file")
+    grading.add_argument('--export', action='store_true',
```
`--export` is broken for v3. Can we remove it for now? Hopefully, we'll get it back at some point.

knrafto replied:
Sure. Should this functionality even be in ok-client?

knrafto commented:
Similar to `--export`, should we remove `--lock` as well? Instead, we should use `ok-lock`, like the 61A repo does.

Sumukh commented:
I think students actually use `--lock` (at least I've told a few students when they want to redo a test suite) - so I'm inclined to keep that one.

knrafto commented:
^ Tested manually too by adding a bogus protocol to one of the demos.

Sumukh commented on client/protocols/backup.py:
```diff
@@ -24,7 +24,7 @@ class BackupProtocol(models.Protocol):
 
     def run(self, messages):
-        if self.args.local or self.args.export or self.args.restore:
+        if self.args.local or self.args.restore:
```
Not in the scope of this PR - but I wonder if there's a better to way to specify when to not run a protocol. We repeat this logic in all protocols.

Maybe all protocols should have a method called `can_run(self)` that returns true or false before we attempt to run them.

Sumukh commented:
LGTM.
